### PR TITLE
feat: outlier options (DHIS2-1369)

### PR DESCRIPTION
**Pull request by martinkrulltott**
Implements [DHIS2-1369](https://jira.dhis2.org/browse/DHIS2-1369)

---

### Key features

1. Added options for the Outliers tab for Scatter charts

---

### Description

Adds all options for the outlier analysis. Features:

1. `Outlier analysis checkbox`: shows/hides all other options
2. `Outlier detection method`: resets the threshold factor to the default value for the selected detection method (`1,5`, `3`, `3` respectively)
3. `Threshold factor`: number input that only allows for values above 0,5
4. `Extreme lines checkbox`: regular checkbox, nothing special
5. `Extreme line % detection`: number input that only allows for values above 0,1 

Example of how this is saved in the store:

` {
  ...
      ui: {
   ...
   options: {
   ...
  outlierAnalysis: {
    enabled: true,
    method: 'IQR',
    thresholdFactor: 1.5,
    extremeLines: {
      enabled: false,
      value: 1 } } } } }`

---

### TODO

- [ ] Add a final help text to the detection method (currently says `TODO` in the ui)

---

### Known issues

- [ ] `Threshold factor` and `Extreme line % detection` should use the `step` prop (0,5 and 0,1 respectively) and the `min` prop (0,5 and 0,1 respectively) once `@dhis2/ui` components support this. https://github.com/dhis2/notes/discussions/240

---

### Screenshots

_default state, unchecked_

image

_default state, checked_

image

### CODE REVIEW

**Review by cooper-joe — APPROVED** [review]
Design LGTM! 👍

File: packages/app/src/components/VisualizationOptions/Options/Outliers.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    tabSectionOptionToggleable,
+    tabSectionOption,
+    tabSectionTitle,
+} from '../styles/VisualizationOptions.style.js'
+import OutlierDetectionMethod from './OutlierDetectionMethod'
+import ExtremeLines from './ExtremeLines'
+import styles from '../styles/Outliers.module.css'
+
+const ENABLED_PROP = 'enabled'
+const METHOD_PROP = 'method'
```
Comment (edoardo) [inline packages/app/src/components/VisualizationOptions/Options/Outliers.js:19]: Looking at [the ticket for adding backend support for the outliers AO option](https://jira.dhis2.org/browse/DHIS2-10079) this should be changed to:

```suggestion
const METHOD_PROP = 'outlierMethod'
```

File: packages/app/src/components/VisualizationOptions/Options/Outliers.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        id: METHOD_MODIFIED_Z_SCORE,
+        label: i18n.t('Modified Z-score'),
+        defaultThreshold: 3,
+    },
+]
+
+const DEFAULT_STATE = {
+    [ENABLED_PROP]: false,
+    [METHOD_PROP]: METHOD_IQR,
+    [THRESHOLD_PROP]: 1.5,
```
Comment (edoardo) [inline packages/app/src/components/VisualizationOptions/Options/Outliers.js:50]: What about `normalizationMethod`?

**Review by edoardo — APPROVED** [review]

File: packages/app/src/components/VisualizationOptions/Options/Outliers.js (hunk 1 above)
Comment (martinkrulltott) [reply packages/app/src/components/VisualizationOptions/Options/Outliers.js:19]: Good find!

File: packages/app/src/components/VisualizationOptions/Options/Outliers.js (hunk 2 above)
Comment (martinkrulltott) [reply packages/app/src/components/VisualizationOptions/Options/Outliers.js:50]: According to @janhenrikoverland we will hardcode the `normalizationMethod` to `XY_RATIO` for now and the prop on the backend is only for future use.

Comment (edoardo) [reply packages/app/src/components/VisualizationOptions/Options/Outliers.js:50]: 👍

File: packages/app/src/components/VisualizationOptions/Options/Outliers.js (hunk 1 above)
Comment (edoardo) [reply packages/app/src/components/VisualizationOptions/Options/Outliers.js:19]: BTW I see Jan's PR has the same constants, maybe import them from analytics?
See https://github.com/dhis2/analytics/pull/831/files/b889bd63c96f25cae7635c6e4f2dc3ea0c15190b..7209db8591c105f53183144dc012615adddb85a0#diff-70166e43854b44e7e94c473a865ff33fe427f9b2d0184a10b62253e36a0144e5

**Comment by martinkrulltott** [comment]
1 Cypress test failing because the user doesn't have any most view visualizations. Merging anyway.

**Pull request merged**